core/src/solve.jl

```diff
@@ -434,14 +431,19 @@ end
 
 function get_error!(pid_control::PidControl, p::Parameters)
     (; basin) = p
-    (; listen_node_id, error) = pid_control
+    (; listen_node_id) = pid_control
+
+    Error = pid_control.error
 
     for i in eachindex(listen_node_id)
         listened_node_id = listen_node_id[i]
         has_index, listened_node_idx = id_index(basin.node_id, listened_node_id)
         @assert has_index "Listen node $listened_node_id is not a Basin."
         target_level = basin.target_level[listened_node_idx]
-        error[i] = target_level - basin.current_level[listened_node_idx]
+        if isnan(target_level)
+            error("No target level specified for listen basin #$listened_node_id.")
+        end
+        Error[i] = target_level - basin.current_level[listened_node_idx]
```

Comment:
```suggestion
    pid_error = pid_control.error

    for i in eachindex(listen_node_id)
        listened_node_id = listen_node_id[i]
        has_index, listened_node_idx = id_index(basin.node_id, listened_node_id)
        @assert has_index "Listen node $listened_node_id is not a Basin."
        target_level = basin.target_level[listened_node_idx]
        if isnan(target_level)
            error("No target level specified for listen basin #$listened_node_id.")
        end
        pid_e
```suggestion
    Error = pid_control.error

    for i in eachindex(listen_node_id)
        listened_node_id = listen_node_id[i]
        has_index, listened_node_idx = id_index(basin.node_id, listened_node_id)
        @assert has_index "Listen node $listened_node_id is not a Basin."
        target_level = basin.target_level[listened_node_idx]
        if isnan(target_level)
            error("No target level specified for listen basin #$listened_node_id.")
        end
        Error[i] = target_level - basin.current_level[listened_node_idx]
```rror[i] = target_level - basin.current_level[listened_node_idx]
```